server/src/main/java/io/littlehorse/server/LHServerListener.java

```diff
@@ -591,6 +591,14 @@ public void searchScheduledWfRun(SearchScheduledWfRunRequest req, StreamObserver
     @Override
     @Authorize(resources = ACLResource.ACL_WORKFLOW, actions = ACLAction.RUN)
     public void runWf(RunWfRequest req, StreamObserver<WfRun> ctx) {
+        if (req.getWfSpecName().equals("")) {
+            throw new LHApiException(Status.INVALID_ARGUMENT, "Missing required parameter 'wf_spec_name'");
+        }
+
+        if (!req.getId().equals("") && !LHUtil.isValidLHName(req.getId())) {
```

Comment:
I wouldn't check for `req.getId().equals("")`. Instead, use the protobuf _optional_ capability which checks for presence of the string. The way to do that is: `req.hasId()`.

I would also suggest adding this:
```
if (req.hasId() && req.getId().equals("")) throw new LHApiException(...);
```